experimental/dds/tree2/src/feature-libraries/sequence-field/rebase.ts

```diff
@@ -867,29 +612,62 @@ function tryRemoveLineageEvent<T>(mark: Attach<T>, revisionToRemove: RevisionTag
 }
 
 /**
- * @returns true iff both reattaches target cells that were affected by the same detach.
- * The target cells may or may not overlap depending on detach index information.
- *
- * Only valid in the context of a rebase (i.e., both marks have the same input context).
- */
-function areRelatedReattaches<T>(baseMark: Reattach<T>, newMark: Reattach<T>): boolean {
-	const baseReference = baseMark.lastDetachedBy ?? baseMark.detachedBy;
-	const newReference = newMark.lastDetachedBy ?? newMark.detachedBy;
-	return baseReference === newReference;
-}
-
-/**
- * @returns true iff `baseMark` attaches nodes in cells whose contents were detached by the same change
- * that conflicts with `newMark`.
- * The target cells may or may not overlap depending on detach index information.
+ * Returns a number N which encodes how the cells of the two marks are aligned.
+ * - If N is zero, then the first cell of `baseMark` is the same as the first cell of `newMark`.
+ * - If N is positive, then the first N cells of `newMark` (or all its cells if N is greater than its length)
+ * are before the first cell of `baseMark`.
+ * - If N is negative, then the first N cells of `baseMark` (or all its cells if N is greater than its length)
+ * are before the first cell of `newMark`.
  */
-function isBaseAttachRelatedToConflictedDetach<T>(
-	baseMark: NewAttach<T> | Reattach<T>,
-	newMark: Detach<T> & Conflicted,
+function compareCellPositions(
 	baseRevision: RevisionTag | undefined,
-): boolean {
-	return (
-		(isActiveReattach(baseMark) && baseMark.detachedBy === newMark.conflictsWith) ||
-		(baseMark.revision ?? baseRevision) === newMark.conflictsWith
+	baseMark: EmptyInputCellMark<unknown>,
+	newMark: EmptyInputCellMark<unknown>,
+	detachOffset: number,
+): number {
+	const baseId = getCellId(baseMark, baseRevision);
+	assert(baseId !== undefined, "baseMark should have cell ID");
+	const newId = getCellId(newMark, undefined);
+	if (baseId.revision === newId?.revision) {
+		return baseId.index - newId.index;
+	}
+
+	if (newId !== undefined) {
+		const offsetInBase = getOffsetAtRevision(baseMark.lineage, newId.revision);
+		if (offsetInBase !== undefined) {
+			// BUG: Cell offsets are not comparable to cell indices.
+			return offsetInBase > newId.index ? offsetInBase - newId.index : -Infinity;
+		}
+	}
+
+	const offsetInNew = getOffsetAtRevision(newMark.lineage, baseId.revision);
+	if (offsetInNew !== undefined) {
+		if (isAttach(baseMark)) {
+			const offset = offsetInNew - detachOffset;
```

Comment:
I don't understand how these two numbers can be related.
Here's a test that (I think) highlights an issue with this codepath:
```
	it("reviveAA ↷ reviveCB => CBAA", () => {
		const reviveAA = Change.revive(0, 2, tag1, 1, rebaseRepair, [
			{ revision: tag2, offset: 1 },
			{ revision: tag3, offset: 1 },
		]);
		const reviveB = composeAnonChanges([
			Change.revive(0, 1, tag2, 0, rebaseRepair),
			Change.revive(0, 1, tag3, 0, rebaseRepair),
		]);
		const expected = Change.revive(2, 2, tag1, 1, rebaseRepair, [
			{ revision: tag2, offset: 1 },
			{ revision: tag3, offset: 1 },
		]);
		const actual = rebase(reviveAA, reviveB);
		assert.deepEqual(actual, expected);
	});
```

Reply:
Good point. This is intended for comparing a roll-forward of a new attach with the lineage entry for its inverse, so we should be gating on `isNewAttach` instead.

Reply:
My issue with this is that `detachOffset` tells us about the gap in the base changeset, but is used as though it is also meaningful in `base.revision`.
I'm guessing you're seeing a scenario in which the base changeset and `base.revision` are indeed equivalent.
If `baseMark` is a new attach then we know for sure it's a roll-forward (i.e., sandwich bread part 2), but it's hard to say for sure whether it may be different from the roll back in a way that makes `detachOffset` inadequate.
I'm happy for us to leave a comment explaining why this is dubious and noting that it'll be fixed when we change to using insertion IDs.